api/rs/slint/mcu.md

```diff
@@ -0,0 +1,405 @@
+
+## Install Toolchain / Hardware Abstraction Layer (HAL)
+
+Writing an application in Rust that runs on a MCU requires the following software components:
+
+* You need to install a Rust toolchain to cross-compile to the target architecture.
+* You need to locate and select the correct HAL crates and drivers, and depend on them in your `Cargo.toml`.
+* You need to install tools for flashing and debugging your code on the device.
+
+Covering these is out of scope for this document, but we recommend reading the [Rust Embedded Book](https://docs.rust-embedded.org/book/),
+as well as the curated list of [Awesome Embedded Rust](https://github.com/rust-embedded/awesome-embedded-rust) for a wide range of different
+crates, tools and training materials. These resources should guide you through the initial setup and often come with "hello world" examples
+to get started with your device.
+
+In addition, Slint requires a nightly version of Rust. This is due to the fact that the support for using a custom global allocator in a bare metal
+environment with `#![no_std]` has not been stabilized yet (see [#51540](https://github.com/rust-lang/rust/issues/51540) or
+[#66741](https://github.com/rust-lang/rust/issues/66741) for tracking issues).
+
+In the following sections we assume that your setup is complete and you have a non-graphical skeleton Rust program running on your MCU.
+
+## Changes to `Cargo.toml`
+
+Start by adding a dependency to the `slint` crate to your `Cargo.toml`:
+
+```toml
+[dependencies]
+... your other dependencies
+
+[dependencies.slint]
+version = "0.3.0"
+default-features = false
+features = ["compat-0.3.0", "unsafe-single-threaded", "libm"]
+
+[build-dependencies]
+slint-build = version = "0.3.0"
+```
+
+The default features of the `slint` create are tailored towards hosted environments and includes the "std" feature. In bare metal environments,
+you need to disable the default features and select at least the `compat-0.3.0` feature (see [this blog post](https://slint-ui.com/blog/rust-adding-default-cargo-feature.html)
+for a longer explanation).
+
+In addition we select two additional features:
+
+ * `unsafe-single-threaded`: Slint internally uses Rust's [`thread_local!`](https://doc.rust-lang.org/std/macro.thread_local.html) macro to store global data.
+   This feature is only available in the Rust Standard Library (std), which is not available in bare-metal environments. As a fallback, the `unsafe-single-threaded`
+   feature will change Slint to use unsafe static for storage. By setting this feature, you guarantee not to use Slint API from a thread other than the main thread,
+   or from interrupt handlers.
+ * `libm`: MCUs often don't provide hardware support for floating point arithmetic. The `libm` feature enables the use of software emulation, with the help of
+   the [libm](https://crates.io/crates/libm) crate.
+
+Finally, we add `slint-build` as a build dependency in order to compile the `.slint` design files to Rust code.
```

Comment:
This sounds like another feature we add, not like an additional crate we need.

Reply:
Well, it's *another* thing in addition to just disabling the default features :-)